src/plans/stepsize.jl

```diff
@@ -1687,3 +1687,193 @@ end
 function get_last_stepsize(step::WolfePowellBinaryLinesearchStepsize, ::Any...)
     return step.last_stepsize
 end
+
+@doc raw"""
+    DistanceOverGradientsStepsize{R<:Real} <: Stepsize
+
+Implements the Riemannian Distance over Gradients (RDoG) stepsize schedule, a learning-rate-free 
+optimization method for Riemannian manifolds introduced by [@dodd2024learning].
+
+This stepsize adapts automatically without hyperparameter tuning by tracking the maximum distance 
+from the initial point and accumulating gradient norms.
+
+# Fields
+
+* `initial_distance::R`: initial distance estimate ``ϵ > 0``
+* `max_distance::R`: maximum distance from initial point ``\bar{r}_t = \max_{s≤t} d(p_0, p_s)``
+* `gradient_sum::R`: accumulated squared gradient norms ``G_t = \sum_{s=0}^t \|∇f(p_s)\|^2``
+* `initial_point`: the initial point ``p_0``
+* `use_curvature::Bool`: whether to use sectional curvature in the stepsize
+* `sectional_curvature_bound::R`: lower bound on sectional curvature ``κ`` (only used if `use_curvature=true`)
+
+The stepsize at iteration ``t`` is computed as:
```

Comment:
I think we usually documented the formulae with the variant without `Stepsize` (so the one with the factory note). That one is considered the public one. This one should document its keywords and refer to the other one.